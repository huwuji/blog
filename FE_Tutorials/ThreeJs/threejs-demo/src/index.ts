// import * as THREE from "three";
import CreateBase from "./utils/createBase";
// import CreateLine from "./utils/line.js";

const { renderer, scene, camera, cube } = CreateBase();

// 剩下的事情就是把它添加到场景中，并调用render（渲染）函数。
// const line = CreateLine(scene, renderer);
// scene.add(line);

// 设置简易动画，然后渲染
function animate() {
  requestAnimationFrame(animate);
  //   cube.rotation.x += 0.01;
  cube.rotation.y += 0.01;
  renderer.render(scene, camera);
}
animate();
