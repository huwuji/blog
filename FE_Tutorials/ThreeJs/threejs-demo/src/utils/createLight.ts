import * as THREE from "three";

/**
 * 添加光源
 */

type LightType = "PointLight" | "RectAreaLight" | "AmbientLight" | "SpotLight";

export default function createLight(
  scene: THREE.Scene,
  type: LightType = "PointLight",
  color?: string
) {
  let light = null;
  switch (type) {
    case "AmbientLight":
      light = new THREE.AmbientLight(color ?? "0xffffff");
      break;
    default:
      light = new THREE[type](color ?? "0xffffff");
      break;
  }
  light.position.set(50, 100, 50);
  scene.add(light);
}
