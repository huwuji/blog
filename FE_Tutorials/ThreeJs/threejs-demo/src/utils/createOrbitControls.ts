/**
 * 创建一个轨道控制器
 *
 */
import { OrbitControls } from "three/examples/jsm/controls/OrbitControls";

export default function controls(
  camera: THREE.Camera,
  renderer: THREE.Renderer
) {
  const controls = new OrbitControls(camera, renderer.domElement);
  return controls;
}
