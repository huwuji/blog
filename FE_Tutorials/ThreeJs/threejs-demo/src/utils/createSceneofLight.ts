import * as THREE from "three";

import setAxesHelper from "./help/setAxesHelper";

/**
 * 设置基础的场景，摄像机，渲染器
 * @returns
 */
export default function () {
  // 定义一个场景
  const scene = new THREE.Scene();
  // 定义一个相机，如下通过构造一个透视相机以及视椎体
  const camera = new THREE.PerspectiveCamera(
    75, //视角角度
    window.innerWidth / window.innerHeight, // 比例理解为生成画布的宽高比
    1,
    1000
  );
  // 定义相机的位置
  camera.position.set(10, 10, 10);

  // 形状，材质  =》网格模型（物体）
  // 形状
  const geometry = new THREE.BoxGeometry(5, 5, 5);
  // MeshLambertMaterial 一种非光泽表面的材质，没有镜面高光。
  const material = new THREE.MeshLambertMaterial({
    color: 0x3f7b9d,
    transparent: true,
    opacity: 0.4,
  });
  const cube = new THREE.Mesh(geometry, material);
  // 设置物体坐标
  cube.position.set(0, 0, 0);
  scene.add(cube);

  // 添加光源
  const light = new THREE.PointLight(0xffffff, 1);
  light.position.set(50, 100, 50);
  scene.add(light);

  camera.lookAt(cube.position);

  // 设置渲染器，渲染输出canvas画布
  const renderer = new THREE.WebGLRenderer();
  // 设置canvas画布的宽高：将输出canvas的大小调整为(width, height)并考虑设备像素比
  renderer.setSize(window.innerWidth / 2, window.innerHeight / 2);

  // 添加axesHelper
  setAxesHelper(scene);

  // 渲染：可以理解为在这个场景下，对这个摄像机的拍摄内容进行渲染
  renderer.render(scene, camera);

  // 将Webgl渲染出的canvas画布添加到dom节点上
  document.body.appendChild(renderer.domElement);

  return {
    scene,
    camera,
    renderer,
    cube,
  };
}
