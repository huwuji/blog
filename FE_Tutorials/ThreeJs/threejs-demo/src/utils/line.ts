import * as THREE from "three";

/**
 *
 * @param {*} scene 场景
 * @param {*} renderer 渲染器
 */
const createLine = function (
  scene: THREE.Scene,
  renderer: THREE.WebGLRenderer
) {
  //对于线条来说，我们能使用的材质只有LineBasicMaterial 或者 LineDashedMaterial。
  const material = new THREE.LineBasicMaterial({ color: 0x0000ff });

  //定义好材质之后，我们需要一个带有一些顶点的geometry（几何体）。
  const points = [];
  points.push(new THREE.Vector3(5, 0, 0));
  points.push(new THREE.Vector3(0, 0, 0));
  points.push(new THREE.Vector3(0, 10, 0));

  const geometry = new THREE.BufferGeometry().setFromPoints(points);

  //注意，线是画在每一对连续的顶点之间的，而不是在第一个顶点和最后一个顶点之间绘制线条（线条并未闭合）。
  // 既然我们已经有了能够画两条线的点和一个材质，那我们现在就可以将他们组合在一起，形成一条线。
  const line = new THREE.Line(geometry, material);
  // 剩下的事情就是把它添加到场景中，并调用render（渲染）函数。
  //   scene.add(line);
  //   renderer.render(scene, camera);
  return line;
};

export default createLine;
