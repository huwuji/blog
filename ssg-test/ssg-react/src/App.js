import { Link } from "react-router-dom";

import logo from './logo.svg';
import './App.css';

function App() {

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <Link to="/myinfo">go myinfo</Link>
      </header>
    </div>
  );
}

export default App;
