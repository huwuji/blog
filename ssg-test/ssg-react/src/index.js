import React from 'react';
import ReactDOM, { hydrate } from 'react-dom';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import './index.css';
import App from './App';
import MyInfo from './router/MyInfo';

import reportWebVitals from './reportWebVitals';

const routers = (
  <BrowserRouter>
    <Routes>
      <Route path="/" element={<App />} />
      <Route path="/myinfo" element={<MyInfo />} />
    </Routes>
  </BrowserRouter>
);

if (window._useSsg) {
  console.log('_useSsg');
  hydrate(routers, document.getElementById('root'));
} else {
  console.log('no_useSsg');

  ReactDOM.render(routers, document.getElementById('root'));
}

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals(console.log);
