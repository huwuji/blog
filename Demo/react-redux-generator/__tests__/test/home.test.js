/**
 * 组件测试
 */
import { createRoot } from "react-dom/client";
import React from "react";
import { renderHook, render, cleanup } from "@testing-library/react";
import Home from "../../src/containers/home/index.js";
import { add } from "../../src/utils";

describe("Comp test", () => {
  afterEach(cleanup);

  it("fn test", () => {
    expect(add(1, 2)).toBe(3);
  });
  // 测试初始快照
  it("测试初始快照", async () => {
    // const { asFragment, getByText } = render(<div>999</div>);
    // console.log("result==", asFragment());
    // expect(getByText(999)).toBeInTheDocument;
    // expect(asFragment(<div>999</div>)).toMatchSnapshot();
    // const { result } = renderHook(() => <div>999</div>);
    // const rooter = createRoot(document.getElementById("root"));
    // rooter.render();
    // console.log("result==", result.current);
  });
});
