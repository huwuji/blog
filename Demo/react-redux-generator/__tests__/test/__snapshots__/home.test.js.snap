// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comp test 测试初始快照 1`] = `
<DocumentFragment>
  <div>
    999
  </div>
</DocumentFragment>
`;
